compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1164,7 +1163,7 @@ impl<'tcx> TyCtxt<'tcx> {
             glob_map: resolutions.glob_map,
             extern_prelude: resolutions.extern_prelude,
             untracked_crate: krate,
-            definitions,
+            definitions: arena.alloc(resolutions.definitions),
```

Comment:
Does this need to be arena-allocated? Can we store the `Definitions` without a reference?